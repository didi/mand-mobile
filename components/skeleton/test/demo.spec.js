import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import Demo2 from './cases/demo2'
import Demo3 from './cases/demo3'
import Demo4 from './cases/demo4'
import {renderToString} from '@vue/server-test-utils'

describe('Skeleton - Demo', () => {
  test(`basic`, () => {
    expect(renderToString(Demo0)).toMatchSnapshot()
  })
  test(`withAvatar`, () => {
    expect(renderToString(Demo1)).toMatchSnapshot()
  })
  test(`switchSkeleton`, () => {
    expect(renderToString(Demo2)).toMatchSnapshot()
  })
  test(`defaultSkeleton`, () => {
    expect(renderToString(Demo3)).toMatchSnapshot()
  })
  test(`combineSkeleton`, () => {
    expect(renderToString(Demo4)).toMatchSnapshot()
  })
})
