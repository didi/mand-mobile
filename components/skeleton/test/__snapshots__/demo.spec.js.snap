// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton - Demo basic 1`] = `
<div class="md-example">
  <div class="md-skeleton">
    <div class="md-skeleton-content">
      <h4 class="md-skeleton-title" style="width:40%;"></h4>
      <div class="md-skeleton-row" style="width:100%;"></div>
      <div class="md-skeleton-row" style="width:100%;"></div>
      <div class="md-skeleton-row" style="width:60%;"></div>
    </div>
  </div>
</div>
`;

exports[`Skeleton - Demo combineSkeleton 1`] = `
<div class="md-example">
  <div class="md-skeleton">
    <div class="md-skeleton-content">
      <h4 class="md-skeleton-title" style="width:36%;"></h4>
      <div class="md-skeleton-row" style="width:40%;"></div>
      <div class="md-skeleton-row" style="width:56%;"></div>
      <div class="md-skeleton-row" style="width:72%;"></div>
      <div class="md-skeleton-row" style="width:60%;"></div>
    </div>
  </div>
</div>
`;

exports[`Skeleton - Demo defaultSkeleton 1`] = `
<div class="md-example">
  <h4></h4>
</div>
`;

exports[`Skeleton - Demo switchSkeleton 1`] = `
<div class="md-example">
  <div class="md-switch skeleton-switch active"></div>
  <fieldset class="md-field">
    <!---->
    <div class="md-field-content">
      <div class="skeleton-item md-skeleton">
        <div class="md-skeleton-avatar"></div>
        <div class="md-skeleton-content">
          <h4 class="md-skeleton-title" style="width:40%;"></h4>
          <div class="md-skeleton-row" style="width:100%;"></div>
          <div class="md-skeleton-row" style="width:60%;"></div>
        </div>
      </div>
      <div class="skeleton-item md-skeleton">
        <div class="md-skeleton-avatar"></div>
        <div class="md-skeleton-content">
          <h4 class="md-skeleton-title" style="width:40%;"></h4>
          <div class="md-skeleton-row" style="width:100%;"></div>
          <div class="md-skeleton-row" style="width:60%;"></div>
        </div>
      </div>
    </div>
    <!---->
  </fieldset>
</div>
`;

exports[`Skeleton - Demo withAvatar 1`] = `
<div class="md-example">
  <div class="md-skeleton">
    <div class="md-skeleton-avatar"></div>
    <div class="md-skeleton-content">
      <h4 class="md-skeleton-title" style="width:40%;"></h4>
      <div class="md-skeleton-row" style="width:100%;"></div>
      <div class="md-skeleton-row" style="width:100%;"></div>
      <div class="md-skeleton-row" style="width:60%;"></div>
    </div>
  </div>
</div>
`;
